vignettes/ChIPpeakAnno.Rmd

```diff
@@ -1,795 +1,1734 @@
+To effectively apply the hypergeometric test, we first need to estimate the total number of binding sites, i.e. `totalTest`. The selection of `totalTest` affects the stringency of the test, with smaller values resulting in a more conservative outcome (larger P-values). For practical guidance on how to choose an appropriate value for `totalTest`, you can refer to this [post](https://stat.ethz.ch/pipermail/bioconductor/2010-November/036540.html). 
+
+In our example, we assume that potential binding regions (which include coding regions and promoter regions) constitute 3% (fairly rough estimation) of the entire genome. Since the example data is derived from chromosome 2, we can estimate the number of total binding sites as `(length of chr2) * 3% / (mean peak length)`.
+```{r, results = "hide", fig.cap = "Venn diagram1 showing overlapping peaks", fig.small = TRUE}
+overlapping_peaks <- findOverlapsOfPeaks(tf1, 
+                                         tf2, 
+                                         tf3, 
+                                         connectedPeaks = "keepAll")
+mean_peak_width <- mean(width(unlist(GRangesList(overlapping_peaks[["all.peaks"]]))))
+
+total_binding_sites <- length(BSgenome.Hsapiens.UCSC.hg38[["chr2"]]) * 0.03 / mean_peak_width
+venn1 <- makeVennDiagram(overlapping_peaks, 
+                         totalTest = total_binding_sites, 
+                         connectedPeaks = "keepAll", 
+                         fill = c("#CC79A7", "#56B4E9", "#F0E442"),
+                         col = c("#D55E00", "#0072B2", "#E69F00"),
+                         cat.col = c("#D55E00", "#0072B2", "#E69F00"))
+```
 
-Bidirectional promoters are the DNA regions located between the transcription start sites (TSS) of two adjacent genes that are transcribed on the opposite directions and often co-regulated by this shared promoter region[@robertson2007].
-Here is an example to find peaks near bi-directional promoters and
-output the percentage of the peaks near bi-directional promoters.
+For the P-values of each peak pair:
+```{r}
+venn1[["p.value"]]
+```
 
-```{r peaksNearBDP16}
-data(myPeakList)
-data(TSS.human.NCBI36)
-seqlevelsStyle(TSS.human.NCBI36) <- seqlevelsStyle(myPeakList)
-annotatedBDP <- peaksNearBDP(myPeakList[1:10,],
-                             AnnotationData=TSS.human.NCBI36,
-                             MaxDistance=5000)
-annotatedBDP$peaksWithBDP
-c(annotatedBDP$percentPeaksWithBDP,
-  annotatedBDP$n.peaks,
-  annotatedBDP$n.peaksWithBDP)
+For the overlapping peak counts:
+```{r}
+venn1[["vennCounts"]]
 ```
 
+The parameter `connectedPeaks = "keepAll"` means that when multiple peaks from different groups overlap, all original peak counts for each group will be displayed in parentheses, while the count of the minimally involved peaks will be displayed outside the parentheses. When `connectedPeaks = "keepFirstListConsistent"` is set, the counts from the first group will be consistently maintained.
```

Comment:
if 'keepAll' is set, when you use findOverlapsOfPeaks and makeVennDiagram, you will see multiple numbers which include all the counts.

Reply:
I have updated the texts to be more clear.